subprojects/docs/src/snippets/buildCache/cacheable-bundle-task/groovy/build.gradle

```diff
@@ -7,10 +7,15 @@ abstract class NpmTask extends DefaultTask {
 
     ListProperty<String> args = project.objects.listProperty(String)
 
+    @Internal
+    protected File bundleJSFile = project.file("$project.buildDir/bundle.js")
```

Comment:
🔴 This changes semantics as the evaluation of `buildDir` would now happen at task instantiation time and could end up observing a stale value. 

An alternative solution would be to use an injected `ProjectLayout`.

Reply:
Ok, I see. Now I try to use injected `ProjectLayout` as you suggest and it's seems it's works. Recheck please